cardano-cli/src/Cardano/CLI/EraBased/Run/Genesis/CreateTestnetData.hs

```diff
@@ -338,40 +341,48 @@ runGenesisCreateTestNetDataCmd
 
     addDRepsToConwayGenesis
       :: [VerificationKey DRepKey]
+      -- \^ The credential of the DReps
       -> [VerificationKey StakeKey]
+      -- \^ The credentials of those that delegate their votes to the dreps. It happens
+      -- to only be stake keys now, but it could be more general in the future.
       -> L.ConwayGenesis L.StandardCrypto
+      -- \^ The genesis data to amend
       -> L.ConwayGenesis L.StandardCrypto
     addDRepsToConwayGenesis dRepKeys stakingKeys conwayGenesis =
-      conwayGenesis
-        { L.cgDelegs = delegs (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
-        , L.cgInitialDReps = initialDReps (L.ucppDRepDeposit $ L.cgUpgradePParams conwayGenesis) dRepKeys
-        }
+      conwayGenesis{L.cgDelegs, L.cgInitialDReps}
      where
-      delegs
-        :: [(VerificationKey StakeKey, VerificationKey DRepKey)]
-        -> ListMap (L.Credential L.Staking L.StandardCrypto) (L.Delegatee L.StandardCrypto)
+      -- The credential, to the drep it delegates to
+      delegs :: [(L.Credential L.Staking L.StandardCrypto, VerificationKey DRepKey)]
       delegs =
-        fromList
-          . map
-            ( bimap
-                verificationKeytoStakeCredential
-                (L.DelegVote . L.DRepCredential . verificationKeyToDRepCredential)
-            )
+        map
+          (first verificationKeytoStakeCredential)
+          (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
+      -- If there are more staking keys than dreps, some dreps don't receive any delegation
+      drepsWithoutDelegation
+        | length stakingKeys >= length dRepKeys = []
+        | otherwise = drop (length stakingKeys) dRepKeys
+
+      minDeposit = L.ucppDRepDeposit $ L.cgUpgradePParams conwayGenesis
+      cgDelegs = fromList $ map (second (L.DelegVote . L.DRepCredential . verificationKeyToDRepCredential)) delegs
+      cgInitialDReps = initialDReps $ drepsWithDelegation ++ drepsWithoutDelegation'
+       where
+        drepsWithDelegation = map (\(stakingCred, drep) -> (drep, Set.singleton stakingCred)) delegs
```

Comment:
This is an issue because it is only a singleton if there are fewer `stakingCred` than `drep`, if not there can be several `stakingCred` for a `drep` (pidgeonhole principle, haha). What seems to be currently happening is that not all delegations are being recorded in the list. This problem was invisible, apparently, because I had a bug in my own code (I now fixed it and pushed a fix to this PR, sorry for that).